test/StateDiffChecker.t.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {Test} from "forge-std/Test.sol";
+import {VmSafe} from "forge-std/Vm.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {LibStateDiffChecker as Checker} from "script/LibStateDiffChecker.sol";
+import {LibStateDiff} from "@eth-optimism-bedrock/scripts/libraries/LibStateDiff.sol";
+
+/// @dev A simple contract to which makes some state modifications for the purpose of testing state diff checking.
```

Comment:
Do you think this is too simple? For example it doesn't verify we handle calls or proxies/delegatecalls correctly. Probably would be good to test a double delegatecall also

Reply:
Sure, I will add some more complexity to the diff.

Reply:
OK, I added new `Callee` and `DelegateCallee` contracts which result in two new diffs.